### PR TITLE
Audio Time Stretching

### PR DESCRIPTION
**Pull request by tomatitito**
I needed to slow down a couple of videos as well as stretch the audio, so that's what I added to moviepy. I added the feature as a new effect in audio/fx. The video/fx/speedx feature now has a flag to indicate wether time stretching should be done, the default is False.

As a bit of a drawback, I used the time stretching algorithm from the librosa project. That means I effectively created another dependency.

- [ ] If this is a bugfix, I have provided code that clearly demonstrates the problem and that works when used with this PR
- [x] I have added a test to the test suite, if necessary
- [x] I have properly documented new or changed features in the documention, or the docstrings
- [ ] I have properly documented unusual changes to the code in the comments around it
- [ ] I have made note of any breaking/backwards incompatible changes

### CODE REVIEW

File: tests/test_fx.py [hunk 1]
```diff
@@ -226,6 +227,15 @@ def test_normalize():
     clip = audio_normalize(clip)
     assert clip.max_volume() == 1
 
+def test_time_stretch():
+    clip = AudioFileClip("media/crunching.mp3")
+
+    clip1=time_stretch(clip, factor=0.5)
+    assert abs(clip1.duration - 2*clip.duration) < 0.1
```
Comment (Overdrivr) [inline tests/test_fx.py]: Could you replace this by [`math.isclose`](https://docs.python.org/3/library/math.html#math.isclose) ? If I'm not mistaken, it should be identical.

File: tests/test_fx.py [hunk 2]
```diff
@@ -226,6 +227,15 @@ def test_normalize():
...
+
+    clip2=time_stretch(clip, factor=1.5)
+    assert abs(1.5*clip2.duration - clip.duration) < 0.1
```
Comment (Overdrivr) [inline tests/test_fx.py]: Same here

File: tests/test_fx.py (hunk 1 above)
Comment (tomatitito) [reply tests/test_fx.py]: Thanks for your feedback! I made the changes.

File: tests/test_fx.py (hunk 2 above)
Comment (tomatitito) [reply tests/test_fx.py]: Same here.

**Review by mondeja — CHANGES_REQUESTED** [review]
Thank you for opening this pull @tomatitito. Before start reviewing the new feature, could you remove the `Pipfile` and the new added gitignore lines? Keep in mind that you can run `git add <filename.ext>` instead of `git add .` when adding files to the commit, so you doesn't need to add new files to gitignore if there aren't strictly needed.

**Comment by tomatitito** [comment]
Thank you, @mondeja for your feedback. Sorry for the Pipfile, I messed up. I fixed some stuff so that the tests are green at least on my machine. I had to call my function qualified, though, as `time_stretch.time_stretch(clip, ...)`, otherwise I would get an error about the module not being callable. It seemed to me though, that all other audio fx functions can be called unqualified, even though I imported mine the same way. Any hints how I could fix this to be more in line with the rest of the moviepy library? 

Also, obviously, the ci pipeline is still not successful. I'll look into this...

**Comment by mondeja** [comment]
> Any hints how I could fix this to be more in line with the rest of the moviepy library?

Just import it in `moviepy.audio.fx.__init__`: `from moviepy.audio.fx.time_stretch import time_stretch`.

**Comment by mondeja** [comment]
Sorry, but we can't depend on more system-level libraries like `libsndfile`, at least for now because it adds a lot of work for us maintaining the installation (and keeping it easiest as possible) for all platforms. Perhaps you can stole the `librosa` `time_stretch` function logic from their codebase?

**Comment by tomatitito** [comment]
Thanks again, @mondeja. I'll try to get rid of librosa.

**Comment by OsaAjani** [comment]
Thank you for your contributions and for reporting issues in this repository. With the release of v2, which introduces significant changes to the codebase and API, we’ve reviewed the backlog of open PRs and issues. Due to the length of the backlog and the likelihood that many of these are either fixed or no longer applicable, we’ve made the decision to close all previous PRs and issues.

If you believe that any of these are still relevant to the current version or if you'd like to reopen a related discussion, please feel free to create a new issue or pull request, referencing the old one.

Thank you for your understanding and continued support!

**Pull request closed without merging**